write.go

```diff
@@ -165,7 +173,12 @@ func writeElement(w dicomio.Writer, elem *Element, opts ...WriteOption) error {
 		return err
 	}
 
-	err = encodeElementHeader(w, elem.Tag, vr, uint32(len(data.Bytes())))
+	length := uint32(len(data.Bytes()))
+	if elem.ValueLength == tag.VLUndefinedLength {
+		length = elem.ValueLength
+	}
```

Comment:
nit: may read a little cleaner imo as 

```suggestion
	length := uint32(len(data.Bytes()))
	if elem.ValueLength == tag.VLUndefinedLength {
		length = tag.VLUndefinedLength
	}
```

Reply:
Sounds good